test/users/npadmana/mpi/test_mpi.chpl

```diff
@@ -0,0 +1,270 @@
+/* A collection of tests/demonstrations of the C-MPI bindings.
+
+These tests are an implementation of the examples here :
+  https://computing.llnl.gov/tutorials/mpi/
+
+Each test is in a separate procedure, each of which ends with an
+ MPI_Barrier (this is especially important since, in some cases, 
+ we only use a subset of the ranks).
+
+*/
+
+/* This initializes MPI by default, you can turn
+that off with the autoInitMPI parameter
+ */
+use MPI; 
+use C_MPI; // Include the C-API, to reduce verbosity of the code.
+use Random;
+
+const requiredSize = 4;
+
+proc main() {
+  if requiredSize != worldSize {
+    writef("Please run with at least %i ranks..\n",requiredSize);
+    MPI_Abort(MPI_COMM_WORLD, 10);
+  }
+
+  hello();
+  point2point();
+  ring();
+  pi();
+  test_scatter();
+  test_structure();
+  test_allgather();
+  test_newcomm();
+
+  MPI_Finalize();
+}
+
+
+proc hello() {
+  /* Simple test of MPI initialization */
+  writef("This is rank %i of %i processes saying Hello, Chapel!\n",worldRank, worldSize);
+  MPI_Barrier(MPI_COMM_WORLD);
+}
+
+/* Simple blocking point-to-point communication */
+proc point2point() {
+
+  if worldRank < 2 {
+    var recvpi : real = 0.0,
+        sendpi : real = 3.1415926;
+    var stat : MPI_Status;
+
+    if worldRank==0 {
+      MPI_Send(sendpi, 1, MPI_DOUBLE, 1, 1, MPI_COMM_WORLD);
```

Comment:
This is related to two comments above. I agree that the C-API is unnecessarily verbose, and we can clean it up. I however don't have a particularly good sense on how one might design a "better" interface. Certainly, trying to get some of the type information automatically would be nice, but so might getting eg. the array information in sends and receives. In fact, one might imagine making these methods on types...

I've put this on a to do list for something to improve in the future.